modules/javafx.graphics/src/main/java/javafx/scene/image/PixelBuffer.java

```diff
@@ -0,0 +1,194 @@
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package javafx.scene.image;
+
+import com.sun.javafx.geom.Rectangle;
+import com.sun.javafx.tk.Toolkit;
+import javafx.geometry.Rectangle2D;
+import javafx.util.Callback;
+
+import java.lang.ref.WeakReference;
+import java.nio.Buffer;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.Objects;
+
+/**
+ * A class to provide a shared pixel data using {@link java.nio.Buffer} for {@code WritableImage}s.
+ * {@code PixelBuffer} supports only {@code PixelFormat.INT_ARGB_PRE}
+ * and {@code PixelFormat.BYTE_BGRA_PRE} pixel formats.
+ * A {@code PixelBuffer} can only be drawn using a {@code WritableImage}.
+ * Multiple {@code WritableImage}s and applications can share a {@code PixelBuffer}.
+ * <p>
+ * Pseudo code to create {@code PixelBuffer}:
+ * <pre>{@code
+ *
+ * // Creating a PixelBuffer using PixelFormat.BYTE_BGRA_PRE
+ * ByteBuffer byteBuffer = ByteBuffer.allocateDirect(width * height * 4);
+ * PixelFormat<ByteBuffer> pixelFormat = PixelFormat.getByteBgraPreInstance();
+ * PixelBuffer pixelBuffer = new PixelBuffer(width, height, byteBuffer, pixelFormat);
+ * Image img = new WritableImage(pixelBuffer);
+ *
+ * // Creating a PixelBuffer using PixelFormat.INT_ARGB_PRE
+ * IntBuffer intBuffer = IntBuffer.allocate(width * height);
+ * PixelFormat<IntBuffer> pixelFormat = PixelFormat.getIntArgbPreInstance();
+ * PixelBuffer pixelBuffer = new PixelBuffer(width, height, intBuffer, pixelFormat);
+ * Image img = new WritableImage(pixelBuffer);
+ * }</pre>
+ *
+ * @since 13
+ */
+public class PixelBuffer<T extends Buffer> {
+
+    private final T buffer;
+    private final int width;
+    private final int height;
+    private final PixelFormat pixelFormat;
+    private ArrayList<WeakReference<WritableImage>> imageRefs = new ArrayList<>();
+
+    /**
+     * Construct a {@code PixelBuffer} from the provided {@link java.nio.Buffer} and {@link PixelFormat}.
+     * The constructor does not allocate memory to store the pixel data, application
+     * must allocate sufficient memory required for the dimensions(x, y) and {@code PixelFormat}.
+     * {@code PixelFormat} must be either {@code PixelFormat.INT_ARGB_PRE} or
+     * {@code PixelFormat.BYTE_BGRA_PRE}, otherwise an exception will be thrown.
+     *
+     * @param width       width dimension of the {@code PixelBuffer}.
+     * @param height      height dimension of the {@code PixelBuffer}.
```

Comment:
dimension is somewhat redundant and doesn't indicate the size of the element (i.e., pixels). Recommend to reword as:

```
     * @param width       width in pixels of this {@code PixelBuffer}
     * @param height      height in pixels of this {@code PixelBuffer}
```

(note that you don't need a period for a simple `@param`)